### PR TITLE
first version of documentation

### PR DESCRIPTION
**Pull request by dorotaphanSiili**

### CODE REVIEW

**Review by w4bremer — CHANGES_REQUESTED** [review]
- Good first version!
- I am not a native speaker, so there are certainly better solutions than my proposals
- Just take the feedback and think about what needs to be changed, I don't expect all my comments to be taken as is :)
- Maybe one or two more screenshots will make it look less technical(boring). e.g. for the QtCreator environment config or so

**Review by w4bremer — CHANGES_REQUESTED** [review]
- Good first version!
- I am not a native speaker, so there are certainly better solutions than my proposals
- Just take the feedback and think about what needs to be changed, I don't expect all my comments to be taken as is :)
- Maybe one or two more screenshots will make it look less technical(boring). e.g. for the QtCreator environment config or so

File: docs/docs/features/qmlplugin.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+```
+where `/io/world` is the module name
+
+If you run your app outside if *Qt creator* you need to add the import paths to runtime. You can either set the QML_IMPORT_PATH environment variable - for ad-hoc import paths (for debugging or testing) or use the QQmlEngine::addImportPath() function for fixed import paths that should always be available. 
+```
+    QGuiApplication app(argc, argv);
+    QQmlApplicationEngine engine;
+    app.addLibraryPath(the-directory);
+```
+When building and running with *qt creator* the may also need to add `QML_IMPORT_PATH` and `QML2_IMPORT_PATH` also for run environment project settings, even if the settings are using build environment variables in which it is present.
```
[the remark this answers is not in the document]
Comment (dorotaphanSiili) [reply docs/docs/features/qmlplugin.md:36]: I'd leave that, it states that normally - within the qt there should be no need to set them, but if it still doesn't work you should add it 🤔

File: docs/docs/features/olink.md [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+ ┃ ┃ ┃ ┣ 📜olinkhello.h
+ ┃ ┃ ┃ ┣ 📜olinkhelloadapter.cpp
+ ┃ ┃ ┃ ┗ 📜olinkhelloadapter.h
+ ...
+```
+
+### Qt apigear - The Network Layer
+
+Together with the olink feature (or monitor feature) you'll see additional folder generated, the 📂hello-world/📂apigear. The 📂olink subfolder contains objects that implement a network layer (based on QtWebSocket library) for the ObjectLink protocol. Those are:
+- OlinkClient - the client, that serves your [interface client adapters](olink#Olink-Client-Adapter).
```
[the remark this answers is not in the document]
Comment (dorotaphanSiili) [reply docs/docs/features/olink.md:72]: OlinkClient holds a message queue for messages while the connection is offline ? 
Yes - but only if connection was once established and object was linked. Same behavior in cpp template.
I guess we should discuss it. I may remove it from documentation? or it should be left as long as the behavior is not changed?

**Review by w4bremer — APPROVED** [review]

**Pull request merged**